google-cloud-error_reporting/lib/google/cloud/error_reporting.rb

```diff
@@ -213,26 +214,40 @@ def self.configure
       # @param [String] service_version A version identifier for running
       #   service.
       #
-      def self.report exception, service_name: nil, service_version: nil, &block
+      def self.report exception, service_name: nil, service_version: nil
         return if Google::Cloud.configure.use_error_reporting == false
 
+        service_name ||= configure.service_name ||
+                         Project.default_service_name
+        service_version ||= configure.service_version ||
+                            Project.default_service_version
+
+        error_event = ErrorEvent.from_exception(exception).tap do |event|
+          event.service_name = service_name
+          event.service_version = service_version
+        end
+
+        yield error_event if block_given?
+
+        default_client.report error_event
+      end
+
+      ##
+      # @private Create a private client to
+      def self.default_client
         unless @@default_client
           project_id = configure.project_id ||
                        Google::Cloud.configure.project_id
           keyfile = configure.keyfile ||
                     Google::Cloud.configure.keyfile
 
-          @@default_client = new project: project_id, keyfile: keyfile
+          @@default_client = AsyncErrorReporter.new(
+            new project: project_id, keyfile: keyfile)
```

Comment:
This is really confusing syntax. Maybe make the calls/parens explicit like you do in middleware.rb. Or create the two objects (ErrorReporting.new and AsyncErrorReporter.new) on separate statements.